pkg/virt-handler/network/pod-network-interface.go

```diff
@@ -0,0 +1,265 @@
+package network
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/020"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/jeevatkm/go-model"
+	libvirtxml "github.com/libvirt/libvirt-go-xml"
+	"github.com/vishvananda/netlink"
+
+	"strings"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/network/cniproxy"
+	"kubevirt.io/kubevirt/pkg/virt-handler/network/utils"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+)
+
+type PodNetworkInterface struct {
+	cniProxy *cniproxy.CNIProxy
+	Name     string
+	IPAddr   net.IPNet
+	Mac      net.HardwareAddr
+}
+
+func (vif *PodNetworkInterface) Plug(domainManager virtwrap.DomainManager) error {
```

Comment:
This function is going to need to be idempotent. When Plug() returns early because of an error, calling the function again should result in it picking up where it left off instead of recreating new interfaces.

From talking with you earlier, I know you're already thinking about this.

My thought is you could store the state of this function as it's being created into a json object written to a shared host mount.  This would basically give you have a local key/value store.  The key is the filename, which could be <namespace>_<domain>.vm, and the value would just be the state.